app/controllers/admin/themes_controller.rb

```diff
@@ -265,6 +265,16 @@ def destroy
     respond_to { |format| format.json { head :no_content } }
   end
 
+  def bulk_destroy
+    themes = Theme.where(id: params[:theme_ids])
+    raise Discourse::InvalidParameters.new(:id) unless themes.present?
+
+    themes.each { |theme| StaffActionLogger.new(current_user).log_theme_destroy(theme) }
```

Comment:
I think this should be in a transaction at least so we are sure that it's either all done or none of it is. Also, can you please double check destroy_all runs AR callbacks (I'm 99% sure it does and delete_all is the one that doesn't, just for my sanity haha) https://github.com/discourse/discourse/blob/47d756f8ba9707f9b9e53eaf69d4b44d28c4e335/app/models/theme.rb#L175-L189